kotti/resources.py

```diff
@@ -801,7 +844,13 @@ def default_get_root(request=None):
             this will be an instance of :class:`~kotti.resources.Document`.
     """
 
-    return DBSession.query(Node).filter(Node.parent_id == None).one()  # noqa
+    # Get the root cache, make one if it doesn't exist yet.
+    try:
+        drc = get_settings()['kotti.default_root_cache']
+    except KeyError:
+        drc = get_settings()['kotti.default_root_cache'] = DefaultRootCache()
+
+    return drc.get_root()
```

Comment:
Isn't this going to clash with multi-threading?  You're handing the SQLAlchemy object from the first thread that called `default_get_root` to all others.